README.md

```diff
@@ -1,111 +1,116 @@
-For example, Porter uses github.com/getporter/packages for publishing our mixin feed.
-
-[Magefile]: https://magefile.org
+# Nomad Mixin for Porter
+
+This is a [Porter](https://porter.sh) mixin for interacting with [Nomad](https://www.nomadproject.io/).
+
+## Installation
+
+todo
+
+## Mixin Syntax
+
+In your porter.yaml file, add the nomad as a mixin:
+
+```yaml
+mixins:
+  - nomad:
+      ...
+```
+
+The Mixin supports the following global settings, which will apply to all your nomad actions, unless overridden in the action
+itself:
+
+```yaml
+mixins:
+  - nomad:
+      address: ""
+      region: ""
+      namespace: ""
+      httpAuth: ""
+
+      # tls
+      caCert: ""
+      caPath: ""
+      clientCert: ""
+      clientKey: ""
+      tlsServerName: ""
+      tlsSkipVerify:
+
+      # acl
+      token: ""
+```
+
+To use the mixin in an Install/Upgrade/Uninstall step, add the "jobs" block to your porter.yaml file (using install here
+an example):
+
+```yaml
+install:
+  - nomad:
+      jobs:
+        - path: nomad/pytechco-redis.nomad.hcl
+        - path: nomad/pytechco-web.nomad.hcl
+        - path: nomad/pytechco-setup.nomad.hcl
+        - dispatch: pytechco-setup
+          meta:
+            budget: "200"
+        - path: nomad/pytechco-employee.nomad.hcl
+```
+
+The job step supports the following 3 actions which can be specified as a list item in the `jobs:` block:
+
+- `path`: runs a nomad job file using the given nomad hcl file.
+- `dispatch`: dispatches a parametrized nomad job using the given job name.
+    - `meta`: (optional): a map of key/value pairs that will be passed as metadata to the nomad job.
+    - `idPrefixTemplate` (optional): a prefix added to dispatched job IDs.
+    - `payload` (optional): base64 encoded string containng the payload to pass to the nomad job. Limited to 65536 bytes.
+- `stop`: stops a nomad job using the given job name.
+    - `purge`: (optional): if set to true, the job will be purged immediately.
+
+Each of the 3 actions above can also take in the following options, which can be passed as arguments to each individual
+nomad run, overriding any global nomad parameters set earlier:
+
+- `address` (string): the address of the nomad server to connect to.
+- `region` (string): the region of the nomad server to connect to.
+- `namespace` (string): the namespace of the nomad server to connect to.
+- `caCert` (string): the path to the CA cert file to use for TLS.
+- `caPath` (string): the path to a directory of CA cert files to use for TLS.
+- `clientCert` (string): the path to the client cert file to use for TLS.
+- `clientKey` (string): the path to the client key file to use for TLS.
+- `tlsServerName` (string): the server name to use as the SNI host when connecting via TLS.
+- `tlsSkipVerify` (bool): disables TLS host verification.
```

Comment:
Setting these in the bundle seems a little weird to me, but I may not be understanding the workflow very well. These values are all specific to a given cluster, so if you ship a bundle with these set, the bundle ends up being specific to that cluster. Also, I'm not sure how/if Porter sandboxes the file system but note that the various TLS-related file paths here will need to be accessible to the Nomad API client.

Reply:
I was thinking that these can be porter parameters or porter credentials as well, to avoid hard coding these settings in a bundle. However, regarding the file system you are right that if the user wants to use a client cert or client key, then they would need to ship these files with the bundle originally, which could potentially be a problem...